simplelink/source/ti/drivers/net/wifi/source/driver.h

```diff
@@ -460,7 +460,7 @@ extern void _SlDrvSleep(_u16 DurationInMsec);
 #endif
 
 #if defined(SL_PLATFORM_MULTI_THREADED)
-extern void * pthread_self(void);
+extern pthread_t pthread_self(void);
```

Comment:
This will not work since this header does not include any headers that would provide the typedef for `pthread_t`.

Reply:
On the contrary - it does work:
```
twister -i -p cc3220sf_launchxl -s samples/net/sockets/http_get/sample.net.sockets.http_get.offload.simplelink
Renaming output directory to /Users/cfriedt/workspace/zephyrproject/zephyr/twister-out.21
INFO    - Using Ninja..
INFO    - Zephyr version: zephyr-v3.2.0-987-gad7ffc405d9a
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /Users/cfriedt/workspace/zephyrproject/zephyr/twister-out/testplan.json
INFO    - JOBS: 8
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - Total complete:    1/   1  100%  skipped:    0, failed:    0
INFO    - 1 test scenarios (1 test instances) selected, 0 configurations skipped (0 by static filter, 0 at runtime).
INFO    - 1 of 1 test configurations passed (100.00%), 0 failed, 0 skipped with 0 warnings in 35.05 seconds
INFO    - In total 0 test cases were executed, 1 skipped on 1 out of total 510 platforms (0.20%)
INFO    - 0 test configurations executed on platforms, 1 test configurations were only built.
INFO    - Saving reports...
INFO    - Writing JSON report /Users/cfriedt/workspace/zephyrproject/zephyr/twister-out/twister.json
INFO    - Writing xunit report /Users/cfriedt/workspace/zephyrproject/zephyr/twister-out/twister.xml...
INFO    - Writing xunit report /Users/cfriedt/workspace/zephyrproject/zephyr/twister-out/twister_report.xml...
INFO    - Run completed
```

See also passing CI
https://github.com/zephyrproject-rtos/zephyr/pull/51784

Reply:
then that is purely by luck because some C files that include this header happen to include POSIX headers.

Reply:
> also that brings the question of, why is this pthread_self forward declaration necessary if the POSIX headers can be included in this scope?

As you are aware, Zephyr POSIX headers **cannot** be portably included in this scope. That is the issue fixed by zephyrproject-rtos/zephyr#51771, which now indirectly depends on this PR.

Let's not make this a chicken-egg problem please.

Reply:
until then, this header should be including `pthread.h` (assuming `pthread.h` is include-able in this scope).

Reply:
in fact, I just removed this `extern pthread_t pthread_self(void);` entirely, and it works.

Reply:
Looks like `pthread.h` is included here:
https://github.com/zephyrproject-rtos/hal_ti/blob/000b944a788b6005d7776198e1348f5c8a657259/simplelink/source/ti/drivers/net/wifi/porting/cc_pal.h#L54